src/OpenLoco/Audio/VehicleChannel.cpp

```diff
@@ -1,18 +1,83 @@
 #include "VehicleChannel.h"
 #include "../Entities/EntityManager.h"
 #include "../Interop/Interop.hpp"
+#include "../Map/TileManager.h"
+#include "../Ui/WindowManager.h"
 #include "../Vehicles/Vehicle.h"
 
 using namespace OpenLoco::Interop;
 
 namespace OpenLoco::Audio
 {
-    static std::pair<SoundId, ChannelAttributes> sub_48A590(const Vehicles::Vehicle2or6* v)
+    static uint8_t getZoomVolumeModifier(uint8_t zoom)
     {
-        registers regs;
-        regs.esi = X86Pointer(v);
-        call(0x0048A590, regs);
-        return { static_cast<SoundId>(regs.eax), { regs.ecx, regs.edx, regs.ebx } };
+        return std::min<uint8_t>(zoom, 2) * 35;
+    }
+
+    static uint8_t getUndergroundVolumeModifier(const Map::Pos3& pos)
+    {
+        if (pos.x != Location::null && Map::validCoords(pos))
+        {
+            return 0;
+        }
+        auto* surface = Map::TileManager::get(pos).surface();
+        if (surface->baseZ() * 4 > pos.z)
+        {
+            return 28;
+        }
+        return 0;
+    }
+
+    static uint8_t getFalloffModifier(int32_t pan)
+    {
+        const auto absPan = std::min(std::abs(pan), 4095);
+
+        uint8_t falloffModifier = 255;
+        // This in theory is the max viewport width/height (might not be valid for modern screens)
+        if (absPan > 2048)
+        {
+            if (absPan > 3072)
+            {
+                falloffModifier = 0;
+            }
+            else
+            {
+                falloffModifier = std::min((3072 - absPan) / 4, 255);
+            }
+        }
+        return falloffModifier;
+    }
+
+    static int32_t calculatePan(const coord_t coord, const int32_t screenSize)
+    {
+        const auto relativePosition = (coord << 16) / std::max(screenSize, 64);
+        return (relativePosition - (1 << 15)) / 16;
+    }
+
+    static std::pair<SoundId, ChannelAttributes> getChannelAttributesFromVehicle(const Vehicles::Vehicle2or6* v)
```

Comment:
Could you add some address comments?
```suggestion
    // 0x0048A590
    static std::pair<SoundId, ChannelAttributes> getChannelAttributesFromVehicle(const Vehicles::Vehicle2or6* v)
```

Reply:
Ahh think that got nuked during my rebase.